### PR TITLE
Adding managed disks support for SDK

### PR DESCRIPTION
**Pull request by epkalyanr**
## Description

The changes in this pull request are related to adding support for Managed disks for the SDK. The managed disks are used for kafka today and we need to provide new fields to the customers to enable them to use the managed disks. We have introduced a new List<DataDisksGroupProperties> in the cluster create parameters which enables the customers to set the data disks properties. The SDK unit tests have been rerecorded since we have introduced a new data model for the datadisks.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by markcowl** [comment]
@shefaliv @nemanja88 @pattipaka   I have assigned you to review this, please review.  Also, we need a firm commitment from your team to have your APIs converted to Swagger within one of the next two sprints (by the end of June), or this PR will not be accepted.  After two sprints, no further changes in Hydra APIs for ARM will be accepted.  Filed an issue here: https://github.com/Azure/azure-sdk-for-net/issues/3058 to track this.

@epkalyanr Need some sign off from those marked as owners for the HDInsight SDK

File: src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs [hunk 1]
```diff
@@ -508,6 +508,13 @@ private static IEnumerable<Role> GetRoleCollection(ClusterCreateParameters clust
                 roles.Add(sandboxHeadNode);
                 return roles;
             }
+
+            List<DataDisksGroupProperties> workernodedatadiskgroups = null;
+            if(clusterCreateParameters.WorkerNodeDataDisksGroups != null && clusterCreateParameters.WorkerNodeDataDisksGroups.Count > 0)
+            {
```
Comment (shefaliv) [inline src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs]: what's this here for? Just set DataDisksGroups = clusterCreateParameters.WorkerNodeDataDisksGroups

File: src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ClusterCreateParameters.cs [hunk 2]
```diff
@@ -194,6 +194,11 @@ public partial class ClusterCreateParameters
         public SecurityProfile SecurityProfile { get; set; }
 
         /// <summary>
+        /// Gets or sets the worker node datadisksgroupproperties for worker nodes. Optional. For Kafka clusters only
```
Comment (shefaliv) [inline src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ClusterCreateParameters.cs]: This is ambiguous. Is it required for Kafka or optional in any case?
If optional for any cluster:
Optional. Gets or sets the worker node DataDisksGroupProperties for worker nodes. Used for Kafka clusters only.

If it's required for Kafka clusters:
Required for Kafka clusters. Gets or sets the worker node DataDisksGroupProperties for worker nodes.

**Review by shefaliv — CHANGES_REQUESTED** [review]

File: src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs (hunk 1 above)
Comment (epkalyanr) [inline src/ResourceManagement/HDInsight/HDInsight/Customizations/ClusterOperations.Create.Customizations.cs]: done !

**Comment by epkalyanr** [comment]
closing this PR as it is not targeting the correct branch. Raised a new PR targeting the HyakNETCore branch.
https://github.com/Azure/azure-sdk-for-net/pull/3077

**Comment by markcowl** [comment]
actually closing the pr

**Pull request closed without merging**